components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/obsrvability/handler/metrics/publisher/MetricReporter.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.micro.integrator.obsrvability.handler.metrics.publisher;
+
+import java.util.Map;
+
+/**
+ * The Metric Reporter is allows one to access the relevant Metric Reporter implementation.
+ */
+public interface MetricReporter {
+    /**
+     * Creates a metric of the provided type with the given metric name,
+     * the metric description and the set of properties that can be used
+     * to identify a metric uniquely and for metric aggregation, metric filtering.
+     *
+     * @param serviceType The service type (Proxy Service, API, Inbound endpoint )
+     *                    for which the metric is going to be instrumented.
+     * @param type        The type of metric (e.g.Counter) that is used in intrumentation.
+     * @param metricHelp  Metric Description
+     * @param properties  Metric labels used to uniquely identify a metric.
+     */
+    public void createMetric(String serviceType, String type, String metricName, String metricHelp,
+                           Map<String, String[]> properties);
+
+    /**
+     * This is called in the handleInit() method of the AbstractExtendedSynapseHandler.
+     * There by the metrics created in the createMetric() method are invoked.
+     */
+    public void initMetrics();
```

Comment:
We can remove the public keyword in interface methods.

Reply:
Fixed